src/v/debug_bundle/tests/debug_bundle_service_test.cc

```diff
@@ -568,3 +568,52 @@ TEST_F_CORO(debug_bundle_service_started_fixture, delete_file_bad_job_id) {
       del_res.assume_error().code(),
       debug_bundle::error_code::job_id_not_recognized);
 }
+
+ss::future<> wait_for_file_to_be_created(
+  const std::filesystem::path& file, const std::chrono::seconds timeout) {
+    const auto start_time = debug_bundle::clock::now();
+    while (debug_bundle::clock::now() - start_time <= timeout) {
+        if (co_await ss::file_exists(file.native())) {
+            co_return;
+        }
+    }
```

Comment:
:fire: 

But please :fire_extinguisher:  with `check_for_io_immediately` or a little sleep.

Reply:
winter is coming 

(warm your CPUs)